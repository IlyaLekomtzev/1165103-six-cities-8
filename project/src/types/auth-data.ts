export type AuthData = {
  email: string;
  password: string;
};

export type UserData = {
  id: number,
  name: string,
  email: string,
  isPro: boolean,
  avatarUrl: string,
  token: string,
};
