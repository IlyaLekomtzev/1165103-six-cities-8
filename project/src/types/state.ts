import { Offer } from './offers';

export type State = {
  city: string;
  sort: string;
  offers: Offer[];
  isLoading: boolean;
  error: string;
};
