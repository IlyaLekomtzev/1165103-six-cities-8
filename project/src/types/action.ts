import {
  ThunkAction,
  ThunkDispatch
} from 'redux-thunk';
import { AxiosInstance } from 'axios';
import { State } from './state';
import * as actions from '../store/action';

export enum ActionType {
  SetCity = 'offers/setCity',
  SetSort = 'offers/setSort',
  SetOffers = 'offers/setOffers',
  SetIsLoading = 'offers/setIsLoading',
  SetError = 'offers/setError',

  SetAuthorizationStatus = 'auth/setAuthorizationStatus',
  SetUserData = 'auth/setUserData',
  RedirectToRoute = 'auth/redirectToRoute',
  RequireLogout = 'auth/requireLogout',
}

type InferValueTypes<T> = T extends {[key: string]: infer U} ? U : never;

export type Actions = ReturnType<InferValueTypes<typeof actions>>;

export type ThunkActionResult<R = Promise<void>> = ThunkAction<R, State, AxiosInstance, Actions>;
export type ThunkAppDispatch = ThunkDispatch<State, AxiosInstance, Actions>;
