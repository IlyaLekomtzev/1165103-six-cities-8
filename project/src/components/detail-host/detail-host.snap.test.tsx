import { render } from '@testing-library/react';
import { User } from '../../types/user';
import DetailHost from './detail-host';

const host: User = {
  id: 1,
  name: 'Test',
  isPro: true,
  avatarUrl: 'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=687&q=80',
};

describe('Component: DetailHost', () => {
  it('should render correctly', () => {
    const {container} = render(<DetailHost host={host} />);
    expect(container).toMatchSnapshot();
  });
});
