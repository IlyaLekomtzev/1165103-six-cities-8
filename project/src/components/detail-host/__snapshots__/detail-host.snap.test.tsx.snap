// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Mistakes should render correctly 1`] = `
<div>
  <h2
    class="property__host-title"
  >
    Meet the host
  </h2>
  <div
    class="property__host-user user"
  >
    <div
      class="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
    >
      <img
        alt="Host avatar"
        class="property__avatar user__avatar"
        height="74"
        src="https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=687&q=80"
        width="74"
      />
    </div>
    <span
      class="property__user-name"
    >
      Test
    </span>
    <span
      class="property__user-status"
    >
      Pro
    </span>
  </div>
</div>
`;
